docs/AQL/grammar/AqlLexer.g4

```diff
@@ -0,0 +1,317 @@
+//
+//  description:  ANTLR4 lexer grammar for Archetype Query Language (AQL)
+//  authors:      Sebastian Iancu, Code24, Netherlands
+//                Teun van Hemert, Nedap, Netherlands
+//                Thomas Beale, Ars Semantica UK, openEHR Foundation Management Board
+//  contributors: This version of the grammar is a complete rewrite of previously published antlr3 grammar,
+//                based on current AQL specifications in combination with other grammars from several AQL implementations.
+//                The openEHR Foundation would like to recognise the following people for their contributions:
+//                  - Chunlan Ma & Heath Frankel, Ocen Health Systems, Australia
+//                  - Bostjan Lah, Better, Slovenia
+//                  - Christian Chevalley, EHRBase, Germany
+//                  - Michael Böckers, Nedap, Netherlands
+//  support:      openEHR Specifications PR tracker <https://specifications.openehr.org/releases/QUERY/open_issues>
+//  copyright:    Copyright (c) 2020 openEHR Foundation
+//  license:      Apache 2.0 License <http://www.apache.org/licenses/LICENSE-2.0.html>
+//
+
+lexer grammar AqlLexer;
+
+channels {
+    COMMENT_CHANNEL
+}
+
+// SKIP
+WS: [ \t\r\n]+ -> skip;
+UNICODE_BOM: (
+    '\uEFBBBF' // UTF-8 BOM
+    | '\uFEFF' // UTF16_BOM
+    | '\u0000FEFF' // UTF32_BOM
+    ) -> skip;
+COMMENT: (
+    SYM_DOUBLE_DASH ' ' ~[\r\n]* ('\r'? '\n' | EOF)
+    | SYM_DOUBLE_DASH ('\r'? '\n' | EOF)
+    ) -> channel(COMMENT_CHANNEL);
+
+// Keywords
+// Common Keywords
+SELECT: S E L E C T ;
+AS: A S ;
+FROM: F R O M ;
+WHERE: W H E R E ;
+ORDER: O R D E R ;
+BY: B Y ;
+DESC: D E S C ;
+DESCENDING: D E S C E N D I N G ;
+ASC: A S C ;
+ASCENDING: A S C E N D I N G ;
+LIMIT: L I M I T ;
+OFFSET: O F F S E T ;
+// other keywords
+DISTINCT: D I S T I N C T ;
+EHR : E H R ;
+VERSION : V E R S I O N ;
+VERSIONED_OBJECT : V E R S I O N E D '_' O B J E C T ;
+LATEST_VERSION : L A T E S T '_' V E R S I O N ;
+ALL_VERSIONS : A L L '_' V E R S I O N S ;
+// deprecated
+TOP: T O P ;
+FORWARD: F O R W A R D ;
+BACKWARD: B A C K W A R D ;
+
+// Operators
+// Containment operator
+CONTAINS : C O N T A I N S ;
+// Logical operators
+AND : A N D ;
+OR : O R ;
+NOT : N O T ;
+EXISTS: E X I S T S ;
+// Arithmetical operators
+ARITHMETIC_OPERATOR: SYM_ASTERISK | SYM_SLASH | SYM_PLUS | SYM_MINUS ;
+// Comparison operators
+COMPARISON_OPERATOR: SYM_EQ | SYM_NE | SYM_GT | SYM_GE | SYM_LT | SYM_LE ;
+LIKE: L I K E ;
+MATCHES: M A T C H E S ;
+
+// functions
+COUNT: C O U N T ;
+MIN: M I N ;
+MAX: M A X ;
+SUM: S U M ;
+AVG: A V G ;
+TERMINOLOGY: T E R M I N O L O G Y ;
+
+//
+PARAMETER: '$' ALPHA_CHAR WORD_CHAR*;
+VARIABLE_ID: IDENTIFIER;
```

Comment:
I don't see the particular need for all these 'aliases' for `IDENTIFIER`. It doesn't help the parser in any way, as deep down they are all the same tokens with a different name. You seem to be using it as some kind of extra documentation, but I suggest keeping it simple here, and just turn 'IDENTIFIER' into a real token, and use that.

Reply:
I would agree with that - if you see differently named tokens, the usual assumption is that they correspond to different match patterns (e.g. names with dots, names without dots, or not allowed to start with a digit, etc etc). Anything that is the same, just use one token rule and id.

Reply:
I had it this way indeed to give an extra doc or hint about the purpose of the token. The second reason is that big parts of the lexer was manually imported (copy) from base_lexer.g4, and there the IDENTIFIER is an fragment.

I'm ok to get rid of all this aliases and use IDENTIFIER, but then what shall I do with 'compatibility' with the base_lexer ?
see https://github.com/openEHR/adl-antlr/tree/master/src/main/antlr/adl2

Reply:
see how was I fixed this: https://github.com/openEHR/specifications-QUERY/pull/15/commits/87c7d0b8932e95564fde3779e4c579e1fb900fb5